### ISSUE TITLE
Is it possible to return a width of rendered text?

### ISSUE THREAD

**Issue by KillaPL** [ruby2d/ruby2d#31]
I would like to use Text#width feature. Is it possible to return something like this from rendered Text element?

I need this, for example, to be able to render buttons. If I want to have a nice button background, I need to be able to know how wide it needs to be.

**Comment by blacktm** [comment]
Anything is possible! 🎩🐇

This should be an easy one. In Simple 2D, we already store both width and height in [`text.c`](https://github.com/simple2d/simple2d/blob/v0.5.1/src/text.c#L46) and in [`text.js`](https://github.com/simple2d/simple2d.js/blob/v0.1.0/src/text.js#L49). Interesting thing to note is that the font and the size determine the width and height, so we won't know what those dimensions are until the text is created. So for example in the native extension, [at initialization](https://github.com/ruby2d/ruby2d/blob/v0.3.0/ext/ruby2d/ruby2d.c#L214) (and after `S2D_CreateText`) we need to store these values in the Ruby object, like with a `r_iv_set(self, "@width", txt->width);` or something.

**Comment by KillaPL** [comment]
Nice! I'm already working on implementation.

**Comment by KillaPL** [comment]
So.

I think I nailed it in mri, and probably in native. Web version is a problem, I created an issue here: https://github.com/simple2d/simple2d.js/issues/1

Also there is a surprising problem: rspec tests for Text#width and Text#height segfaults, even though testcard-style tests works fine. This is probably due to my c code.

Before I show any code it's probably best to first merge https://github.com/ruby2d/ruby2d/pull/25 if and when it's ready, since it touches similiar areas and there would be conflicts, and I'd like to solve the js problem first.
